Subject: [PATCH] Fix for wakelock being held during bootup and idle scenario.

On bootup of device, if device is kept idle without unlocking the
screen the TCXO does not happen due to "ActivityManager-Sleep"
partial wake_lock being held.

Fix releases the wakelock irrespective of the result of code flow
in checkReadyForSleepLocked() function.

diff --git a/services/core/java/com/android/server/am/ActivityStackSupervisor.java b/services/core/java/com/android/server/am/ActivityStackSupervisor.java
index 3de2036..3b42374 100644
--- a/services/core/java/com/android/server/am/ActivityStackSupervisor.java
+++ b/services/core/java/com/android/server/am/ActivityStackSupervisor.java
@@ -3173,6 +3173,9 @@ void goingToSleepLocked() {
         applySleepTokensLocked(false /* applyToStacks */);
 
         checkReadyForSleepLocked(true /* allowDelay */);
+        if (mGoingToSleep.isHeld()) {
+            mGoingToSleep.release();
+        }
     }
 
     void prepareForShutdownLocked() {
@@ -3283,9 +3286,6 @@ void checkReadyForSleepLocked(boolean allowDelay) {
 
         removeSleepTimeouts();
 
-        if (mGoingToSleep.isHeld()) {
-            mGoingToSleep.release();
-        }
         if (mService.mShuttingDown) {
             mService.notifyAll();
         }
